codegen/smithy-aws-kotlin-codegen/src/test/kotlin/software/amazon/smithy/kotlin/codegen/aws/protocols/RpcV2CborTest.kt

```diff
@@ -98,4 +120,29 @@ class RpcV2CborTest {
         """.replaceIndent("        ")
         getFooMethod.shouldContainOnlyOnceWithDiff(expectedHeaderMutation)
     }
+
+    @Test
+    fun testEventStreamContentTypeHeaders() {
+        val ctx = model.newTestContext("CborExample")
+
+        val generator = RpcV2Cbor()
+        generator.generateProtocolClient(ctx.generationCtx)
+
+        ctx.generationCtx.delegator.finalize()
+        ctx.generationCtx.delegator.flushWriters()
+
+        val serializer = ctx.manifest.expectFileString("/src/main/kotlin/com/test/serde/PutFooStreamingOperationSerializer.kt")
+
+        // Event stream messages should have `:content-type: application/cbor`
+        val encodeMessage = serializer.lines("private fun encodePutFooStreamingPublishEventsEventMessage(input: PublishEvents): Message = buildMessage {", "}")
+        encodeMessage.shouldContainOnlyOnceWithDiff(
+            """
+            addHeader(":content-type", HeaderValue.String("application/cbor"))
+            """.trimIndent(),
+        )
+
+        // Event stream requests should have Content-Type=application/vnd.amazon.eventstream
+        val serializeBody = serializer.lines("    override suspend fun serialize(context: ExecutionContext, input: PutFooStreamingRequest): HttpRequestBuilder {", "}")
+        serializeBody.shouldContainOnlyOnceWithDiff("""builder.headers.setMissing("Content-Type", "application/vnd.amazon.eventstream")""")
```

Comment:
Question: This is coming from line 252 of `HttpBindingProtocolGenerator.kt` right?

Reply:
yes